airflow/dag_processing/processor.py

```diff
@@ -574,7 +574,7 @@ def execute_callbacks(
                 if isinstance(request, TaskCallbackRequest):
                     self._execute_task_callbacks(dagbag, request)
                 elif isinstance(request, SlaCallbackRequest):
-                    self.manage_slas(dagbag.dags.get(request.dag_id))
+                    self.manage_slas(dagbag.dags[request.dag_id])
```

Comment:
Error here:
```
airflow/dag_processing/processor.py:577: error: Argument 1 to "manage_slas" of "DagFileProcessor" has incompatible type "Optional[DAG]"; expected "DAG"
                        self.manage_slas(dagbag.dags.get(request.dag_id))
```
Not entirely sure if there is a case when a task would have an SLA callback but not have its DAG in the DagBag. If it's possible, I can certainly modify `manage_slas()` typing and raise an exception for a missing DAG if this is a possible scenario.

Reply:
SLA doesn’t make sense without a DAG, so this is better IMO. It’s good to fail here if things go wrong.

Reply:
Yeah. They basically can't go wrong here or we are in a really deep mess.

Reply:
Yup, the change should be good